content/docs/user-guide/checkpoints.md

```diff
@@ -0,0 +1,388 @@
+training our model. In your terminal, run
+`dvc stage add -n download -d download.py -o data/MNIST python download.py`.
+This will create a _dvc.yaml_ file in your root directory.
+
+The command we just ran creates a stage called _download_ with a dependency on
+the _download.py_ file and it should output any results to the _data/MNIST_
+directory.
+
+Inside of the yaml file, you should see the following code.
+
+```yaml
+stages:
+  download:
+    cmd: python download.py
+    deps:
+      - download.py
+    outs:
+      - data/MNIST
+```
+
+Now we need to add a stage for training our model within a DVC pipeliene. We'll
+do that with the following command:
+`dvc stage add -n train -d data/MNIST -d train.py -c model.pt --plots-no-cache predictions.json -p seed,lr,weight_decay --live dvclive python train.py`
+
+## Enabling DVC pipelines for checkpoint experiments
+
+After running the command above to setup your _train_ stage, your _dvc.yaml_
+should have the following code.
+
+```yaml
+stages:
+  download:
+    cmd: python download.py
+    deps:
+      - download.py
+    outs:
+      - data/MNIST
+  train:
+    cmd: python train.py
+    deps:
+      - data/MNIST
+      - train.py
+    params:
+      - seed
+      - lr
+      - weight_decay
+    outs:
+      - model.pt:
+          checkpoint: true
+    plots:
+      - predictions.json:
+          cache: false
+    live:
+      dvclive:
+        summary: true
+        html: true
+```
+
+The checkpoints need to be enabled in DVC at the pipeline level. The
+`-c / --checkpoint` option of the `dvc stage add` command defines the checkpoint
+file or directory. This will create a new stage with checkpoints enabled, but
+does not update existing stages.
+
+The rest of the command `dvc stage add` sets up our dependencies for running the
+training code, which parameters we want to track (which are defined in the
+_params.yaml_), some configurations for our plots and showing the training
+metrics.
+
+We'll manually add a slight modification to the way our plots are handled. Add
+the following lines to your _dvc.yaml_. This just changes the way our results
+are shown after we've run some experiments.
+
+```yaml
+plots:
+  - predictions.json:
+      cache: false
+      template: confusion
+      x: actual
+      y: predicted
+```
+
+Now that you know how to enable checkpoints in a DVC pipeline, let's move on to
+setting up checkpoints in your code.
+
+## Registering checkpoints in your code
+
+Take a look at the _train.py_ file and you'll see how we train a convolutional
+neural network to classify handwritten digits. The main area of this code most
+relevant to checkpoints is when we iterate over the training epochs.
+
+This is where DVC will be able to track our metrics over time and where we will
+add our checkpoints to give us the points in time with our model that we can
+switch between.
+
+To run this code, run the `dvc exp run` command and it will start training the
```

Comment:
Also, it might be better to introduce the code changes before running `dvc exp run`. Otherwise, a user won't see the checkpoints.

Reply:
The next section is "Running experiments". So, you can just remove `dvc exp run` from the current one. Otherwise, users will try to run it and will be confused by the result.